packages/backend/src/links/links.service.ts

```diff
@@ -1,26 +1,45 @@
-import { Injectable } from '@nestjs/common'
-import { CreateLinkDto } from './dto/create-link.dto'
-import { UpdateLinkDto } from './dto/update-link.dto'
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateLinkDto } from './dto/create-link.dto';
+import { UpdateLinkDto } from './dto/update-link.dto';
+import { PrismaService } from 'nestjs-prisma';
+import { Prisma } from '@prisma/client';
 
 @Injectable()
 export class LinksService {
+  constructor(private readonly prisma: PrismaService) {}
   create(createLinkDto: CreateLinkDto) {
-    return 'This action adds a new link'
+    try {
+      return this.prisma.link.create({ data: createLinkDto });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          console.log('This slug violates the unique constraint, meaning it already exists!');
+        }
+      }
```

Comment:
throw a HttpException (like bad request) here, otherwise the endpoint will return with an empty response